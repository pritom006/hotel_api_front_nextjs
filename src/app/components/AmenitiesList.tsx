// src/app/components/AmenitiesList.tsx
import React from 'react';

interface AmenitiesListProps {
  amenities: string[];
}

const AmenitiesList: React.FC<AmenitiesListProps> = ({ amenities }) => {
  return (
    <div className="mt-6">
      <h2 className="text-lg font-semibold mb-2">Popular Amenities</h2>
      <div className="flex flex-wrap gap-2">
        {amenities.map((amenity, index) => (
          <span key={index} className="px-3 py-1 bg-gray-200 rounded-full text-sm">
            {amenity}
          </span>
        ))}
      </div>
    </div>
  );
};

export default AmenitiesList;
