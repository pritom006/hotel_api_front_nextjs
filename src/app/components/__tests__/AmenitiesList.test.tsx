import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom'; // Provides additional matchers like toBeInTheDocument
import AmenitiesList from '../AmenitiesList';

describe('AmenitiesList', () => {
  const mockAmenities = [
    'Parking',
    'Kitchen',
    'Washer',
    'Dryer',
    'Outdoor Space',
    'Barbecue grill',
    'WiFi', // More than 6 amenities
  ];

  it('renders amenities with the correct icons', () => {
    render(<AmenitiesList amenities={mockAmenities} />);

    // Check if the first 6 amenities are rendered with their icons
    expect(screen.getByText(/Parking/i)).toBeInTheDocument();
    expect(screen.getByText('🚗')).toBeInTheDocument();

    expect(screen.getByText(/Kitchen/i)).toBeInTheDocument();
    expect(screen.getByText('🍳')).toBeInTheDocument();

    expect(screen.getByText(/Washer/i)).toBeInTheDocument();
    expect(screen.getByText('🧺')).toBeInTheDocument();

    expect(screen.getByText(/Dryer/i)).toBeInTheDocument();
    expect(screen.getByText('👕')).toBeInTheDocument();

    expect(screen.getByText(/Outdoor Space/i)).toBeInTheDocument();
    expect(screen.getByText('🌳')).toBeInTheDocument();

    expect(screen.getByText(/Barbecue grill/i)).toBeInTheDocument();
    expect(screen.getByText('🔥')).toBeInTheDocument();
  });

  it('does not render more than 6 amenities', () => {
    render(<AmenitiesList amenities={mockAmenities} />);

    // Only the first 6 amenities should be rendered
    expect(screen.queryByText('WiFi')).not.toBeInTheDocument(); // 7th amenity should not be displayed
  });

  it('displays the "See all property amenities" link when there are more than 6 amenities', () => {
    render(<AmenitiesList amenities={mockAmenities} />);
    
    // Check if the link to see all amenities is displayed
    expect(screen.getByText(/See all property amenities/i)).toBeInTheDocument();
  });

  it('does not display the "See all property amenities" link when there are 6 or fewer amenities', () => {
    const fewAmenities = ['Parking', 'Kitchen', 'Washer', 'Dryer', 'Outdoor Space'];
    render(<AmenitiesList amenities={fewAmenities} />);
    
    // Check that the link is not displayed for fewer than 6 amenities
    expect(screen.queryByText(/See all property amenities/i)).not.toBeInTheDocument();
  });

  it('matches the snapshot', () => {
    const { asFragment } = render(<AmenitiesList amenities={mockAmenities} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
