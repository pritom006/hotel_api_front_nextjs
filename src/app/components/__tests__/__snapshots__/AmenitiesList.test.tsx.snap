// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AmenitiesList matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="mb-8"
  >
    <h2
      class="text-xl font-semibold mb-4"
    >
      Popular amenities
    </h2>
    <div
      class="grid grid-cols-2 md:grid-cols-3 gap-4 mb-4"
    >
      <div
        class="flex items-center gap-2"
      >
        <span>
          🚗
        </span>
        <span>
          Parking
        </span>
      </div>
      <div
        class="flex items-center gap-2"
      >
        <span>
          🍳
        </span>
        <span>
          Kitchen
        </span>
      </div>
      <div
        class="flex items-center gap-2"
      >
        <span>
          🧺
        </span>
        <span>
          Washer
        </span>
      </div>
      <div
        class="flex items-center gap-2"
      >
        <span>
          👕
        </span>
        <span>
          Dryer
        </span>
      </div>
      <div
        class="flex items-center gap-2"
      >
        <span>
          🌳
        </span>
        <span>
          Outdoor Space
        </span>
      </div>
      <div
        class="flex items-center gap-2"
      >
        <span>
          🔥
        </span>
        <span>
          Barbecue grill
        </span>
      </div>
    </div>
    <a
      class="text-blue-600 hover:underline"
      href="#"
    >
      See all property amenities →
    </a>
  </div>
</DocumentFragment>
`;
